tt_eager/tt_dnn/op_library/moreh_sum_backward/moreh_sum_backward_op.cpp

```diff
@@ -100,18 +106,24 @@ operation::ProgramWithCallbacks MorehSumBackward::create_program(
 
 Tensor moreh_sum_backward(
     const Tensor &output_grad,
-    const Tensor &input,
+    const std::optional<const Tensor> input,
     std::optional<std::variant<int64_t, std::vector<int64_t>>> dim,
     const bool keep_batch_dim,
     const std::optional<const Tensor> input_grad,
     const MemoryConfig &input_grad_mem_config,
     std::optional<const DeviceComputeKernelConfig> compute_kernel_config) {
 
-    std::vector<int64_t> dims = get_dim(dim, input.get_legacy_shape().rank());
+    TT_FATAL((input.has_value() || input_grad.has_value()), "either input or input_grad must have a value");
+    uint32_t rank = input.has_value() ? input->get_legacy_shape().rank() : input_grad->get_legacy_shape().rank();
+    std::vector<int64_t> dims = get_dim(dim, rank);
     std::sort(dims.begin(), dims.end());
 
-    std::vector<Tensor> output_tensors = {Tensor(operation::get_workers_for_op_output({output_grad, input}))};
-    auto kernel_config_val = init_device_compute_kernel_config(input.device()->arch(), compute_kernel_config, MathFidelity::HiFi4);
+    std::vector<Tensor> input_tensors = { output_grad };
+    if (input) {
+        input_tensors.emplace_back(*input);
+    }
```

Comment:
```suggestion
```
I don't think `input` needs to be included in `input_tensors`. Since the third slot of `launch_op` is optional_input_tensors, `input` can be placed there.